runtime/compiler/runtime/RuntimeAssumptions.hpp

```diff
@@ -45,6 +45,16 @@ class TR_PreXRecompile;
 class TR_RedefinedClassPicSite;
 class TR_UnloadedClassPicSite;
 
+/**
+ * @brief Enum to describe the result of a Class Chain Validation (CCV).
+ */
+enum CCVResult
```

Comment:
Does something like this work on all platforms: `enum CCVResults : uint8_t`
This would eliminate the need for casting.

Reply:
I tried it out; it didn't work. The compiler will accept the syntax, but it will complain about invalid conversion without the casts.

Reply:
I am curious, on which platform it didn't work? We use this for JITServer code compiled with gcc

Reply:
```
diff --git a/runtime/compiler/runtime/RuntimeAssumptions.hpp b/runtime/compiler/runtime/RuntimeAssumptions.hpp
index 293b1333fd..176b02d3c4 100644
--- a/runtime/compiler/runtime/RuntimeAssumptions.hpp
+++ b/runtime/compiler/runtime/RuntimeAssumptions.hpp
@@ -48,7 +48,7 @@ class TR_UnloadedClassPicSite;
 /**
  * @brief Enum to describe the result of a Class Chain Validation (CCV).
  */
-enum CCVResult
+enum CCVResult : uint8_t
    {
    notYetValidated,
    success,
@@ -160,8 +160,8 @@ class TR_PersistentClassInfo : public TR_Link0<TR_PersistentClassInfo>
    int32_t getNameLength()                       { return _nameLength; }
    virtual void setNameLength(int32_t length)            { _nameLength = length; }
 
-   void setCCVResult(CCVResult result) { _ccvResult = static_cast<uint8_t>(result); }
-   CCVResult getCCVResult() const { return static_cast<CCVResult>(_ccvResult); }
+   void setCCVResult(CCVResult result) { _ccvResult = result; }
+   CCVResult getCCVResult() const { return _ccvResult; }
 
    private:
```
```
In file included from /root/jit/compiler/env/J9SharedCache.hpp:34:0,
                 from /root/jit/compiler/env/VMJ9.h:45,
                 from /root/jit/compiler/compile/AOTClassInfo.hpp:29,
                 from /root/jit/compiler/compile/J9Compilation.hpp:43,
                 from /root/jit/compiler/compile/Compilation.hpp:26,
                 from /root/jit/omr/compiler/codegen/OMRCodeGenerator.hpp:45,
                 from /root/jit/omr/compiler/x/codegen/OMRCodeGenerator.hpp:34,
                 from /root/jit/omr/compiler/x/amd64/codegen/OMRCodeGenerator.hpp:36,
                 from /root/jit/compiler/codegen/J9CodeGenerator.hpp:36,
                 from /root/jit/compiler/x/codegen/J9CodeGenerator.hpp:26,
                 from /root/jit/compiler/x/amd64/codegen/J9CodeGenerator.hpp:41,
                 from /root/jit/compiler/codegen/CodeGenGC.cpp:32:
/root/jit/compiler/runtime/RuntimeAssumptions.hpp: In member function 'CCVResult TR_PersistentClassInfo::getCCVResult() const':
/root/jit/compiler/runtime/RuntimeAssumptions.hpp:164:44: error: invalid conversion from 'uint8_t {aka unsigned char}' to 'CCVResult' [-fpermissive]
    CCVResult getCCVResult() const { return _ccvResult; }
                                            ^~~~~~~~~~
```
```
root@ba22b76deb8d:~/jit# gcc --version
gcc (GCC) 7.3.0
Copyright (C) 2017 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
```

Reply:
What is the declaration of `_ccvResult` above? The compiler complains that about a conversion from `uint8_t` to `CCVResult`, but we should already have:
`CCVResult _ccvResult;`

Reply:
oh, `_ccvResult` is a `uint8_t`. I guess that's why there's a problem?

Reply:
yeah looks like changing `_ccvResult` to `CCVResult` fixes the build issue.

Reply:
I will have to try on other platforms though before making the change here.

Reply:
Changes made.